### PR TITLE
Doc(next): Update documentation for handling App router part

### PR DESCRIPTION
**Pull request by arimet**

### CODE REVIEW

File: docs/NextJs.md [hunk 1]
```diff
@@ -39,16 +39,18 @@ Add the `react-admin` npm package, as well as a data provider package. In this e
...
 ## Creating The Admin App Component
 
-Next, create a `components` directory inside `src`, and an admin App component in `src/components/AdminApp.jsx`:
+Next, create a `components` directory inside `src`, and an admin App component in `src/components/AdminApp.tsx`:
 
 ```jsx
-// in src/components/AdminApp.jsx
-"use client"; // only needed if you choose App Router
+// in src/components/AdminApp.tsx
+"use client"; // remove it if you choose Pages Router
```
Comment (djhi) [inline docs/NextJs.md:53]:
```suggestion
"use client"; // remove this line if you choose Pages Router
```

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by djhi — APPROVED** [review]

File: docs/NextJs.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ npx create-next-app@latest
 ```
 
 A prompt will asks you some questions, feel free to choose answers according to your needs. 
-This tutorial assumes you're using an `src` folder, so answer 'Yes' to the 5th question. As for the App Router, you can choose to use it or not, this tutorial will explain how to use both.
+This tutorial assumes you're using an `src` folder, so answer 'Yes' to the 5th question. As for the App Router, you can choose to use it or not, this tutorial will explain how to use both. (For new applications, Next.js recommends using the App Router.)
```
Comment (adguernier) [inline docs/NextJs.md:26]:
```suggestion
This tutorial assumes you're using a `src` folder, so answer `Yes` to the 5th question. As for the App Router, you can choose to use it or not, this tutorial will explain how to use both. (For new applications, Next.js recommends using the App Router).
```

File: docs/NextJs.md [hunk 3]
```diff
@@ -157,16 +159,95 @@ SUPABASE_URL="https://MY_INSTANCE.supabase.co"
 SUPABASE_SERVICE_ROLE="MY_SERVICE_ROLE_KEY"
 ```
 
-**Tip**: This example uses the **service role key** here and not the anonymous role. This allows mutations without dealing with authorization. **You shouldn't do this in production**, but use the [Supabase authorization](https://supabase.com/docs/guides/auth) feature instead.
+**Tip**: This example uses the **service role key** here and not the anonymous role. This allows mutations without dealing with authorization (You may have to modify the safety policies.). **You shouldn't do this in production**, but use the [Supabase authorization](https://supabase.com/docs/guides/auth) feature instead.
```
Comment (adguernier) [inline docs/NextJs.md:162]:
```suggestion
**Tip**: This example uses the **service role key** here and not the anonymous role. This allows mutations without dealing with authorization (You may have to modify the safety policies). **You shouldn't do this in production**, but use the [Supabase authorization](https://supabase.com/docs/guides/auth) feature instead.
```

File: docs/NextJs.md [hunk 4]
```diff
@@ -157,16 +159,95 @@ SUPABASE_URL="https://MY_INSTANCE.supabase.co"
...
+/!\ The file name is important: it must be `route.ts` in the App Router and `[[...slug]].ts` in the Pages Router.
+
+From this point on, the logic for handling is different depending on the router.
+
+### App Router
+
+For more information about routes handler with the App Router, see [the official documentation](https://nextjs.org/docs/app/building-your-application/routing/route-handlers).
+
+```tsx
+// In src/app/api/admin/[...slug]/route.ts
```
Comment (adguernier) [inline docs/NextJs.md:178]:
```suggestion
// in src/app/api/admin/[...slug]/route.ts
```

File: docs/NextJs.md [hunk 5]
```diff
@@ -157,16 +159,95 @@ SUPABASE_URL="https://MY_INSTANCE.supabase.co"
...
+
+### Pages Router
+
+For more information about routes handler with the Pages Router, see [the official documentation](https://nextjs.org/docs/pages/building-your-application/routing/api-routes#optional-catch-all-api-routes).
+
+
 This API route redirects all calls from the react-admin app to the Supabase CRUD API:
 
 ```tsx
+// In src/pages/api/admin/[[...slug]].ts
```
Comment (adguernier) [inline docs/NextJs.md:250]:
```suggestion
// in src/pages/api/admin/[[...slug]].ts
```

File: docs/NextJs.md [hunk 6]
```diff
@@ -200,19 +281,34 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
 **Tip**: Some of this code is really PostgREST-specific. The `prefer` header is required to let PostgREST return one record instead of an array containing one record in response to `getOne` requests. The `Content-Range` header is returned by PostgREST and must be passed down to the client. A proxy for another CRUD API will require different parameters.
 
+### Dataprovider
+
 Finally, update the react-admin data provider to use the Supabase adapter instead of the JSON Server one. As Supabase provides a PostgREST endpoint, we'll use [`ra-data-postgrest`](https://github.com/raphiniert-com/ra-data-postgrest):
 
 ```sh
+npm install @raphiniert/ra-data-postgrest // or
```
Comment (adguernier) [inline docs/NextJs.md:289]:
```suggestion
npm install @raphiniert/ra-data-postgrest 
# or
```

File: docs/NextJs.md [hunk 7]
```diff
@@ -77,7 +79,7 @@ export default AdminApp;
 
 This is a minimal configuration to render CRUD pages for users, posts and comments. React-admin will guess the fields to display in the list and edition pages based on the API response. 
 
-**Tips**: If you choose App Router, do not forget to add [the `"use client"` directive](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive).
+**Tips**: If you choose Pages Router, do not forget to remove [the `"use client"` directive](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive).
```
Comment (fzaninotto) [inline docs/NextJs.md:82]: Do we need this tip? If they use the Pages router, the `"use client"` directive does nothing.

File: docs/NextJs.md [hunk 8]
```diff
@@ -157,16 +159,95 @@ SUPABASE_URL="https://MY_INSTANCE.supabase.co"
...
 - Pages Router: `src/pages/api/admin/[[...slug]].ts`
- 
+
+/!\ The file name is important: it must be `route.ts` in the App Router and `[[...slug]].ts` in the Pages Router.
+
+From this point on, the logic for handling is different depending on the router.
+
+### App Router
+
+For more information about routes handler with the App Router, see [the official documentation](https://nextjs.org/docs/app/building-your-application/routing/route-handlers).
```
Comment (fzaninotto) [inline docs/NextJs.md:175]: I'd put this comment after the code snippet

File: docs/NextJs.md [hunk 9]
```diff
@@ -157,16 +159,95 @@ SUPABASE_URL="https://MY_INSTANCE.supabase.co"
...
+    return new Response(data, {
+        status: 200,
+        headers,
+    });
+}
+```
+
+### Pages Router
+
+For more information about routes handler with the Pages Router, see [the official documentation](https://nextjs.org/docs/pages/building-your-application/routing/api-routes#optional-catch-all-api-routes).
```
Comment (fzaninotto) [inline docs/NextJs.md:244]: I'd put this comment after the code snippet

File: docs/NextJs.md [hunk 10]
```diff
@@ -200,19 +281,34 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
 **Tip**: Some of this code is really PostgREST-specific. The `prefer` header is required to let PostgREST return one record instead of an array containing one record in response to `getOne` requests. The `Content-Range` header is returned by PostgREST and must be passed down to the client. A proxy for another CRUD API will require different parameters.
 
+### Dataprovider
```
Comment (fzaninotto) [inline docs/NextJs.md:284]:
```suggestion
### Data Provider
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**